src/database/migration/file/migration_version_8.ts

```diff
@@ -0,0 +1,26 @@
+import {SQLiteDatabase} from 'react-native-sqlite-storage';
+
+import ChatSchema from '../../schema/ChatSchema';
+import Migration from './migration.types';
+
+const TABLE_NAME = ChatSchema.getTableName();
+
+class MigrationVersion8 implements Migration {
+  up = async (db: SQLiteDatabase): Promise<void> => {
+    const upQuery = `ALTER TABLE ${TABLE_NAME}
+      ADD COLUMN attachment_json TEXT DEFAULT NULL;`;
+
+    console.log(`===== MIGRATING: ${TABLE_NAME.toLocaleUpperCase()} TABLE =====`);
+    await db.executeSql(upQuery);
+    console.log(`===== DONE MIGRATING: ${TABLE_NAME.toLocaleUpperCase()}  TABLE =====`);
+  };
+
+  down = async (db: SQLiteDatabase): Promise<void> => {
+    console.log(`===== DOWNGRADING: ${TABLE_NAME.toLocaleUpperCase()} TABLE =====`);
+    const downQuery = `ALTER TABLE ${TABLE_NAME} DROP COLUMN attachment_json`;
+    await db.executeSql(downQuery);
+    console.log(`===== DONE DOWNGRADING: ${TABLE_NAME.toLocaleUpperCase()}  TABLE =====`);
+  };
+}
+
+export default new MigrationVersion8();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The migration script looks good overall, but there's a potential issue with the `down` function. SQLite does not support dropping a column directly using `ALTER TABLE DROP COLUMN`. You might need to create a new table without the `attachment_json` column, copy data from the old table to the new one, and then rename the new table.

```diff
  down = async (db: SQLiteDatabase): Promise<void> => {
    console.log(`===== DOWNGRADING: ${TABLE_NAME.toLocaleUpperCase()} TABLE =====`);
-20:     const downQuery = `ALTER TABLE ${TABLE_NAME} DROP COLUMN attachment_json`;
+20:     const tempTableName = `${TABLE_NAME}_temp`;
+21:     const createTempTableQuery = `CREATE TABLE ${tempTableName} AS SELECT * FROM ${TABLE_NAME};`;
+22:     const dropOldTableQuery = `DROP TABLE ${TABLE_NAME};`;
+23:     const renameTableQuery = `ALTER TABLE ${tempTableName} RENAME TO ${TABLE_NAME};`;
    await db.executeSql(createTempTableQuery);
+25:     await db.executeSql(dropOldTableQuery);
+26:     await db.executeSql(renameTableQuery);
    console.log(`===== DONE DOWNGRADING: ${TABLE_NAME.toLocaleUpperCase()}  TABLE =====`);
  };
```
Please note that this approach will lose all indexes, triggers, and constraints related to the dropped column. If you have any of these, you'll need to recreate them manually.


<!-- This is an auto-generated comment by OSS CodeRabbit -->